beacon_node/network/src/subnet_service/tests/mod.rs

```diff
@@ -338,110 +339,83 @@ mod attestation_service {
         assert_eq!(subnet_id1, subnet_id2);
 
         // submit the subscriptions
-        attestation_service
+        subnet_service
             .validator_subscriptions(vec![sub1, sub2].into_iter())
             .unwrap();
 
         // Unsubscription event should happen at slot 2 (since subnet id's are the same, unsubscription event should be at higher slot + 1)
-        // Get all events for 1 slot duration (unsubscription event should happen after 2 slot durations).
-        let events = get_events(&mut attestation_service, None, 1).await;
-        matches::assert_matches!(
-            events[..3],
-            [
-                SubnetServiceMessage::Subscribe(_any1),
-                SubnetServiceMessage::EnrAdd(_any3),
-                SubnetServiceMessage::DiscoverPeers(_),
-            ]
-        );
-
         let expected = SubnetServiceMessage::Subscribe(Subnet::Attestation(subnet_id1));
 
-        // Should be still subscribed to 2 long lived and up to 1 short lived subnet if both are
-        // different.
-        if !attestation_service.is_subscribed(
-            &subnet_id1,
-            attestation_subnets::SubscriptionKind::LongLived,
-        ) {
-            // The index is 3*subnets_per_node (because we subscribe + discover + enr per long lived
-            // subnet) + 1
-            let index = 3 * subnets_per_node;
-            assert_eq!(expected, events[index]);
-            assert_eq!(
-                attestation_service.subscription_count(),
-                subnets_per_node + 1
-            );
+        if subnet_service.is_subscribed(&Subnet::Attestation(subnet_id1)) {
```

Comment:
Looks like this is checking `short lived subscriptions`?

Reply:
Not sure how it's possible to hit the `else` block given that the subscription should have happened. is this a timing thing?

Reply:
I think here we're worried about the long-lived subscriptions. 

If the subnet chosen, happens to be the same one we are permanently subscribed to, then we don't create a subscription or unsubscription and is_subscribed() i think only returns short-lived subscriptions